openpype/hosts/maya/api/lib.py

```diff
@@ -1326,8 +1326,14 @@ def is_visible(node,
             return False
 
     if displayLayer:
-        # Display layers set overrideEnabled and overrideVisibility on members
-        if cmds.attributeQuery('overrideEnabled', node=node, exists=True):
+        # If a display layer is connected, it'll override the value of
+        # "overrideVisibility" which ends up returning 0, so we need to query
+        # the connected display layer.
+        display_layers = cmds.listConnections(node, type="displayLayer")
+        if display_layers:
+            if not cmds.getAttr("{}.visibility".format(display_layers[0])):
+                return False
+        else:
```

Comment:
Sorry, can you explain why if it's connected - why does it end up returning `0`?

---

This is not the correct fix. This will break if a node has a connection to a display layer for any others reason, even if it's **not** connected to its enabled state or if it's connected to its `.overrideVisibility` attribute.

It would already be better if this were:
```python
cmds.listConnections(f"{node}.overrideVisibility", type="displayLayer", source=True, destination=False)
```

So that we're much closer to actually checking the _actual_ connection we want. Otherwise this might falsely think it's driven by a display layer if e.g. a node has an attribute that "drives" the display layer instead or any other custom logic is involved a studio might have.

The code snippet I shared here is better, but it does leave me to wonder why this needs "fixing" to begin with? Where does the bug come from?

Reply:
> Sorry, can you explain why if it's connected - why does it end up returning 0?

I'm not sure why. You can test it yourself in Maya.

Reply:
Any reproducable on when it occurs? :)

Reply:
For reference, this works fine:
```python
layer = cmds.createDisplayLayer(name="layer1", number=1, nr=True)

cube = cmds.polyCube(constructionHistory=False)[0]
cmds.editDisplayLayerMembers(layer, cube)
override = cmds.getAttr(cube + ".overrideEnabled")
print(override)
print("visibility", cmds.getAttr(f"{cube}.overrideVisibility"))
cmds.setAttr(layer + ".visibility", False)
print("visibility", cmds.getAttr(f"{cube}.overrideVisibility"))
cmds.setAttr(layer + ".visibility", True)
print("visibility", cmds.getAttr(f"{cube}.overrideVisibility"))
```

Reply:
You are right about the logic issue being something else. The issue was that when `overrideVisibility` was True, `is_visible` would return False, but that should be inverted to return False when `overrideVisibility` is False.